services/algorithms/dqn/__init__.py

```diff
@@ -0,0 +1,33 @@
+import sys
+from collections import deque
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from services.algorithms import AlgorithmBASE
+from services.constants import *
+
+class DQNBASE(AlgorithmBASE):
+  def __init__(self, **kwargs):
+    super().__init__(**kwargs)
+    # self.target_model = keras.models.clone_model(self.model) # fixed Q targets
+    # self.target_model.set_weights(self.model.get_weights())
+
+    self.replay_buffer = deque(maxlen=kwargs[MAX_REPLAY_BUFFER_LENGTH]) # change to a circular buffer if replay length gets long
+
+    self.buffer_wait_steps = kwargs[BUFFER_WAIT_STEPS]
+
+  def sample_experiences(self):
```

Comment:
I like to name internal methods with leading double underscores, can you please put those back?

Reply:
There was an error inheritance for the methods with double underscores.